projects/github-actions/repo-gardening/src/tasks/triage-issues/index.js

```diff
@@ -11,6 +14,112 @@ const sendSlackMessage = require( '../../utils/slack/send-slack-message' );
+	if ( repoLabels.length === 0 ) {
+		debug( 'triage-issues: No labels found in the repository. Aborting OpenAI request.' );
+		return suggestions;
+	}
+
+	const prompt = `Here is the issue title. It is the most important part of the text you must analyse:
+${ title }
+
+Here is the issue body:
+${ body }
+
+Here are the existing labels and their descriptions:
+${ repoLabels.join( ', ' ) }
+
+Analyze the issue and suggest relevant labels. Rules:
+- Don't search for and add more labels if the existing labels match your initial assessment.
+- Use only existing labels provided.
+- Include 1 '[Feature Group]' label.
+- Include 1 to 3 '[Feature]' labels.
+- Briefly explain each label choice in 1 sentence.
+- Format your response as a JSON object with 'labels' and 'explanations' keys.
+
+Example response format:
+{
+    "labels": ["[Product] WordPress.com", "[Feature Group] User Interaction & Engagement", "[Feature] Comments"],
+    "explanations": {
+        "[Feature Group] User Interaction & Engagement": "The issue involves how users interact with the platform.",
+        "[Feature] Comments": "Specifically, it's about the commenting functionality."
+    }
+}`;
+
+	const response = await sendOpenAiRequest( prompt );
+
+	let parsedResponse;
+	try {
+		parsedResponse = JSON.parse( response );
+	} catch ( error ) {
+		debug(
+			`triage-issues: OpenAI did not send back the expected JSON-formatted response. Error: ${ error }`
+		);
+		return suggestions;
+	}
+
+	debug( `triage-issues: OpenAI response: ${ JSON.stringify( parsedResponse ) }` );
+	const { labels, explanations } = parsedResponse;
+
+	// Display the explanations in the action logs.
+	const explanationString = Object.entries( explanations ).join( ' // ' );
+	debug( `triage-issues: OpenAI suggested the following labels: ${ explanationString }` );
+
+	if ( ! Array.isArray( labels ) ) {
+		return suggestions;
+	}
+
+	return { labels, explanations };
+}
+
+/**
+ * Check if the issue has the "[Experiment] Automated labeling" label.
+ *
+ * @param {GitHub} octokit    - Initialized Octokit REST client.
+ * @param {string} owner      - Repository owner.
+ * @param {string} repo       - Repository name.
+ * @param {string} number     - Issue number.
+ * @param {string} action     - Action that triggered the event ('opened', 'reopened', 'labeled').
+ * @param {object} eventLabel - Label that was added to the issue.
+ *
+ * @return {Promise<boolean>} Promise resolving to boolean.
+ */
+async function hasTestLabel( octokit, owner, repo, number, action, eventLabel ) {
+	// Check existing labels.
+	const labels = await getLabels( octokit, owner, repo, number );
+	if ( labels.includes( '[Experiment] Automated labeling' ) ) {
+		return true;
+	}
+
+	// Next, check if the current event was a [Experiment] Automated labeling label being added.
+	if (
+		'labeled' === action &&
+		eventLabel.name &&
+		'[Experiment] Automated labeling' === eventLabel.name
+	) {
+		return true;
+	}
```

Comment:
Since this task is triaged whenever issues are opened, but also when they are labeled, it can happen that this task will be triggered just when the "[Experiment] Automated labeling" label will be added. That means `getLabels()` will not include that label yet (because it is not yet part of the issue), but the new label will be the one that's added to the issue. This is what this logic is meant to catch. 

Does that make sense?

Reply:
Why would `getLabels()` not include the label that was just added?

Reply:
Looking through the history, I see you originally added code like this in #30100 in response to [this event](https://github.com/Automattic/wp-calypso/issues/75728#event-9000969272).

But there were a bunch of label manipulations there within the same minute. I suspect the re-addition was in a run triggered by [one of the earlier labels being added](https://github.com/Automattic/wp-calypso/issues/75728#event-9000957285) that saw one of the intermediate states rather than by the addition of `[Pri] Normal`.

Reply:
You're right. I did a quick test, and the returned labels include the one that was just added:
https://github.com/jeherve/test-actions/issues/56

I pushed 398e8236e2339994de840afbb96a7bb557560a59 and it seems to work well, like in this example PR:
https://github.com/jeherve/jetpack/issues/154